examples/installation/build.rb

```diff
@@ -0,0 +1,241 @@
+    elsif not Dir.exist? '../../build/Realm.xcframework'
+      raise 'Missing XCFramework to test in ../../build'
+    end
+
+  else
+    usage
+  end
+end
+
+def build_app(platform, method, static)
+  archive_path = "#{Dir.pwd}/out.xcarchive"
+  FileUtils.rm_rf archive_path
+
+  build_args = ['clean', 'archive', '-archivePath', archive_path]
+  case platform
+  when 'ios'
+    build_args += ['-sdk', 'iphoneos', '-destination', 'generic/platform=iphoneos']
+  when 'tvos'
+    build_args += ['-sdk', 'appletvos', '-destination', 'generic/platform=appletvos']
+  when 'watchos'
+    build_args += ['-sdk', 'watchos', '-destination', 'generic/platform=watchos']
+  when 'osx'
+    build_args += ['-sdk', 'macosx', '-destination', 'generic/platform=macOS']
+  when 'catalyst'
+    build_args += ['-destination', 'generic/platform=macOS,variant=Mac Catalyst']
+  end
+  build_args += ['CODE_SIGN_IDENTITY=', 'CODE_SIGNING_REQUIRED=NO', 'AD_HOC_CODE_SIGNING_ALLOWED=YES']
+
+  case method
+  when 'cocoapods'
+    sh 'xcodebuild', '-workspace', 'CocoaPods.xcworkspace', '-scheme', 'App', *build_args
+
+  when 'carthage'
+    sh 'xcodebuild', '-project', 'Carthage.xcodeproj', '-scheme', 'App', *build_args
+
+  when 'spm'
+    sh 'xcodebuild', '-project', 'SwiftPackageManager.xcodeproj', '-scheme', 'App', *build_args
+
+  when 'xcframework'
+    if static
+      sh 'xcodebuild', '-project', 'Static/StaticExample.xcodeproj', '-scheme', 'StaticExample', *build_args
+    else
+      sh 'xcodebuild', '-project', 'XCFramework.xcodeproj', '-scheme', 'App', *build_args
+    end
+  end
+end
+
+def validate_build(static)
+  has_frameworks = Dir["out.xcarchive/Products/Applications/**/Frameworks/*.framework"].length != 0
+  if has_frameworks and static
+    raise 'Static build configuration has embedded frameworks'
+  elsif not has_frameworks and not static
+    raise 'Dyanmic build configuration is missing embedded frameworks'
+  end
+end
+
+def test(platform, method, linkage = 'dynamic')
+  # Because we only have one target Xcode will choose to build us as a static
+  # library when using spm
+  static = linkage == 'static' || method == 'spm'
+  if static
+    ENV['REALM_BUILD_STATIC'] = '1'
+  else
+    ENV.delete 'REALM_BUILD_STATIC'
+  end
+
+  puts "Testing #{method} for #{platform}"
+
+  download_realm(platform, method, static)
+  build_app(platform, method, static)
+  validate_build(static)
+end
+
+if ARGV[0] == 'test-all'
+  platforms = ['ios', 'osx', 'tvos', 'watchos', 'catalyst']
+  if /15\..*/ =~ XCODE_VERSION
+    platforms += ['visionos']
+  end
+
+  for platform in platforms
+    for method in ['cocoapods', 'carthage', 'spm', 'xcframework']
+      next if platform == 'catalyst' && method == 'carthage'
+      next if platform == 'visionos' && method != 'spm'
+      test platform, method, 'dynamic'
+    end
+
+    test platform, 'cocoapods', 'static' unless platform == 'visionos'
+  end
+
+  test 'ios', 'xcframework', 'static'
```

Comment:
For historical reasons we ship an iOS-specific static framework (iOS 7 didn't support dynamic frameworks, and we just kept shipping it even after dropping iOS 7). These changes generalize the build so that the static framework isn't iOS-specific but we're still only shipping it for iOS.